src/gpu/utils.rs

```diff
@@ -62,3 +62,23 @@ pub fn get_core_count(d: Device) -> GPUResult<usize> {
         }),
     }
 }
+
+use std::fs::{remove_file, File};
+use std::path::Path;
+use std::sync::Mutex;
+pub const FILE_NAME: &str = "/tmp/bellman_gpu.lock";
+lazy_static::lazy_static! {
+    static ref IS_ME : Mutex<bool> = Mutex::new(false);
+}
+pub fn gpu_acquire() {
+    *IS_ME.lock().unwrap() = true;
+    File::create(FILE_NAME).unwrap();
+}
+pub fn gpu_release() {
+    *IS_ME.lock().unwrap() = false;
+    remove_file(FILE_NAME).unwrap();
+}
+pub fn gpu_is_available() -> bool {
+    // Either taken by me or not taken by somebody else
+    *IS_ME.lock().unwrap() || !Path::new(FILE_NAME).exists()
```

Comment:
I might miss something. If a process having a lock (via `gpu_acquire()`) crashes, the lock file will still be around. A call to `gpu_is_available()` would then always return true. Would the caller of this library be responsible for making sure the lock file doesn't exist when it starts up?

Reply:
@vmx You are right, if the process crashes, that "signaling" file would still be around. I think its reasonable to expect the process with more priority to manage that GPU acquisition permission. I'll think more about it. @dignifiedquire how does that sound to you?